WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,606 @@
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        if (mSearchView != null) {
+            mSearchView.clearFocus();
+        }
+        ActivityUtils.hideKeyboard(this);
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        if (!StringUtils.equals(query, mSearchQuery)) {
+            submitSearch(query, true);
+        }
+        return true;
+    }
+
+    private void configureSearchView() {
+        mSearchView = findViewById(R.id.search_view);
+
+        // don't allow the SearchView to be closed
+        mSearchView.setOnCloseListener(new SearchView.OnCloseListener() {
+            @Override public boolean onClose() {
+                return true;
+            }
+        });
+
+        mSearchView.setOnQueryTextListener(this);
+    }
+
+    private void showEmptyView(boolean show) {
+        if (!isFinishing()) {
+            TextView txtEmpty = findViewById(R.id.text_empty);
+            txtEmpty.setVisibility(show ? View.VISIBLE : View.GONE);
+            if (show) {
+                boolean isEmpty = mSearchQuery == null || mSearchQuery.length() < MIN_SEARCH_QUERY_SIZE;
+                if (isEmpty) {
+                    String message = getString(R.string.stock_media_picker_initial_empty_text);
+                    String subMessage = getString(R.string.stock_media_picker_initial_empty_subtext);
+                    String link = "<a href='https://pexels.com/'>Pexels</a>";
+                    String html = message
+                                  + "<br /><br />"
+                                  + "<small>" + String.format(subMessage, link) + "</small>";
+                    txtEmpty.setMovementMethod(WPLinkMovementMethod.getInstance());
+                    txtEmpty.setText(Html.fromHtml(html));
+                } else {
+                    txtEmpty.setText(R.string.stock_media_picker_empty_results);
+                }
+            }
+        }
+    }
+
+    private void showProgress(boolean show) {
+        if (!isFinishing()) {
+            findViewById(R.id.progress).setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void submitSearch(@Nullable final String query, boolean delayed) {
+        mSearchQuery = query;
+
+        if (query == null || query.length() < MIN_SEARCH_QUERY_SIZE) {
+            mAdapter.clear();
+            showEmptyView(true);
+            return;
+        }
+
+        if (delayed) {
+            mHandler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    if (StringUtils.equals(query, mSearchQuery)) {
+                        submitSearch(query, false);
+                    }
+                }
+            }, 500);
+        } else {
+            fetchStockMedia(query, 1);
+        }
+    }
+
+    private void fetchStockMedia(@Nullable String searchQuery, int page) {
+        if (mIsFetching || !NetworkUtils.checkConnection(this)) {
```

Comment:
I think this should be something like this:

```
if ((mIsFetching && page != 1) || !NetworkUtils.checkConnection(this)) {
```

This way we only block loading more pages if we are already fetching, but we still make the first page requests because that'll mean the search query changed. The issue is actually fairly easy to reproduce. Start searching for `dog` and as soon as the progress bar shows up (which means the fetch started) add `s` to the end to make it `dogs`. The fetch will never complete because the search query is changed, but we can't start a new search because it'll always think we are already fetching.

P.S: I thought about raising this issue when I recommended moving the `mIsFetching` check to this method. I probably should have said so. This was also one of the reasons I wanted all the checks to happen in one place. Because we could explain the check process through comments like below. It's not that important though and I am fine with the way it is right now.

```
// We should always fetch the first page
// We should only load more pages if we are not already fetching anything
```